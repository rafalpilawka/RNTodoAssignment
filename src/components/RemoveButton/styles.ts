import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  text: {color: 'white', fontSize: 16, fontWeight: 'bold'},
  button: {
    backgroundColor: 'red',
    borderRadius: normalize(50),
    width: normalize(30),
    height: normalize(30),
    justifyContent: 'center',
    alignItems: 'center',
  },
});
