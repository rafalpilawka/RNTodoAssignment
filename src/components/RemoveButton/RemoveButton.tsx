import React, {ReactElement} from 'react';
import {Pressable, Text} from 'react-native';
import {styles} from 'src/components/RemoveButton/styles';

export type PropsT = {
  onPress(): void;
};

export const RemoveButton = ({onPress}: PropsT): ReactElement => {
  return (
    <Pressable style={styles.button} onPress={onPress}>
      <Text style={styles.text}>x</Text>
    </Pressable>
  );
};
