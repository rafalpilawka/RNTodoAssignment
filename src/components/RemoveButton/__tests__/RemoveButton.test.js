import React from 'react';
import 'react-native';
import {Pressable} from 'react-native';
import {shallow} from 'enzyme';
import {RemoveButton} from 'src/components/RemoveButton/RemoveButton';
import renderer from 'react-test-renderer';

const props = {
  onPress: jest.fn(),
};

describe('RemoveButton', () => {
  let component;
  beforeEach(() => {
    component = shallow(<RemoveButton {...props} />);
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<RemoveButton {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should call action', () => {
    const button = component.find(Pressable);
    button.simulate('press');
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
