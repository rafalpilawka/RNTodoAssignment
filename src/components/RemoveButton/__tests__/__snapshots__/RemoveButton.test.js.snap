// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveButton should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 100,
      "height": 60,
      "justifyContent": "center",
      "width": 60,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    x
  </Text>
</View>
`;
